src/Common/src/Interop/Unix/libssl/SslConnectionInfo.cs

```diff
@@ -190,5 +202,32 @@ private static HashAlgorithmType MapHashAlgorithmType(Interop.libssl.DataHashAlg
                     return HashAlgorithmType.None;
             }
         }
+        
+        private static int GetHashAlgorithmBitSize(Interop.libssl.DataHashAlgorithm mac)
+        {
+            switch (mac)
+            {
+                case Interop.libssl.DataHashAlgorithm.SSL_MD5:
+                    return MD5_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA1:
+                    return SHA1_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST94:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST89MAC:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA256:
+                    return SHA256_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA384:
+                    return SHA384_HashKeySize;
+
+                default:
+                    return 0;
```

Comment:
apart from the macs mentioned there is one more mac which is AEAD ..  which is as per the explanation in Openssl is not is real mac .
AEAD is only used with *GCM Cipher Algorithm.. if *GCM encryption algorthim (cipher algo) is used, it typically does both the work encryption and hashing.

 "TLSv1.2 adds a new class of ciphers using AEAD modes such as GCM, which is a single crypto operation that does both encryption and integrity protection, instead of having separate encryption and HMAC e.g. AES128-CBC plus HMAC-SHA1. Mac=AEAD means really no HMAC is needed because of AEAD."

So I think returning some default thing would be the better idea.